pkg/scalers/postgres_scaler.go

```diff
@@ -0,0 +1,204 @@
+package scalers
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	_ "github.com/lib/pq"
+	"k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	pgMetricName            = "num"
+	defaultPostgresPassword = ""
+
+	pgMetricType = "External"
+)
+
+type postGRESScaler struct {
+	metadata   *postGRESMetadata
+	connection *sql.DB
+}
+
+type postGRESMetadata struct {
+	connStr  string
+	userName string
+	password string
+	host     string
+	port     string
+	query    string
+	dbName   string
+	sslmode  string
+}
+
+var postgresLog = logf.Log.WithName("postgres_scaler")
+
+// NewPostgresScaler creates a new postgres scaler
+func NewPostgresScaler(resolvedEnv, metadata, authParams map[string]string) (Scaler, error) {
+	meta, err := parsePostgresMetadata(resolvedEnv, metadata, authParams)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing postgres metadata: %s", err)
+	}
+
+	conn, err := getConnection(meta)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing postgres connection: %s", err)
+	}
+	return &postGRESScaler{
+		metadata:   meta,
+		connection: conn,
+	}, nil
+}
+
+func parsePostgresMetadata(resolvedEnv, metadata, authParams map[string]string) (*postGRESMetadata, error) {
+	meta := postGRESMetadata{}
+	meta.host = "foo"
+
+	if val, ok := metadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := authParams["connStr"]; ok {
+		meta.connStr = val + "?sslmode=disable"
```

Comment:
//TODO: I ran into an issue with the way PG defaults its SSL connections. Might be worth considering whether to add sslmode to connStr